ftdc_test.go

```diff
@@ -161,3 +163,47 @@ func TestReadPathIntegration(t *testing.T) {
 		})
 	})
 }
+
+func TestRoundtrip(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 300*time.Second)
+	defer cancel()
+
+	collectors := createCollectors()
+	for _, collect := range collectors {
+		t.Run(collect.name, func(t *testing.T) {
+			tests := createTests()
+			for _, test := range tests {
+				t.Run(test.name, func(t *testing.T) {
+					if test.numStats == 0 {
+						t.Skip("does not have stats or has random stats")
+					}
+					if test.randStats && !strings.Contains(collect.name, "dynamic") {
+						t.Skip("non-dynamic collectors cannot handle random stats")
+					}
```

Comment:
I'd move these both to before line 176, and just use continue rather than t.Skip, which I would generally reserve for cases where you have a test that is broken and you know it so you want to skip it rather than fail it directly.